doc/gamate.sgml

```diff
@@ -0,0 +1,159 @@
+cc65 C compiler.
+</abstract>
+
+<!-- Table of contents -->
+<toc>
+
+<!-- Begin the document -->
+
+<sect>Overview<p>
+
+This file contains an overview of the Gamate runtime system as it comes
+with the cc65 C compiler. It describes the memory layout, Gamate specific header
+files, available drivers, and any pitfalls specific to that platform.
+
+Please note that Gamate specific functions are just mentioned here, they are
+described in detail in the separate <url url="funcref.html" name="function
+reference">. Even functions marked as "platform dependent" may be available on
+more than one platform. Please see the function reference for more
+information.
+
+
+<sect>Binary format<p>
+
+The standard binary output format generated by the linker for the Gamate target
+is a cartridge image with header. It is of course possible to change this
+behaviour by using a modified startup file and linker config.
+
+Note: the first two bytes of the header contain a checksum that must be inserted
+by an external program. Such an utility is provided in util/gamate/gamate-fixcart.c
+
+<sect>Platform specific header files<p>
+
+Programs containing Gamate specific code may use the <tt/gamate.h/ header file.
+
+
+<sect1>Hardware access<p>
+
+The following pseudo variables declared in the <tt/gamate.inc/ include file do
+allow access to hardware located in the address space.
+
+<descrip>
+
+</descrip><p>
+
+
+
+<sect>Loadable drivers<p>
+
+All drivers must be statically linked because no file I/O is available.
+The names in the parentheses denote the symbols to be used for static linking of the drivers.
+
+
+<sect1>Graphics drivers<p>
+
+No TGI graphics drivers are currently available for the Gamate.
+
+
+<sect1>Extended memory drivers<p>
+
+No extended memory drivers are currently available for the Gamate.
+
+
+<sect1>Joystick drivers<p>
+
+<descrip>
+
+  <tag><tt/gamate-stdjoy.joy (gamate_stdjoy)/</tag>
+  A joystick driver for the standard two buttons joypad is available.
+
+</descrip><p>
+
+
+<sect1>Mouse drivers<p>
+
+No mouse drivers are currently available for the Gamate.
+
+
+<sect1>RS232 device drivers<p>
+
+No serial drivers are currently available for the Gamate.
+
+
+
+<sect>Limitations<p>
+
+<itemize>
+<item>When using the C-compiler, keep in mind that only 0x200 bytes RAM in total
+can be used for variables and the runtime stack.
+</itemize>
+
+<sect1>Disk I/O<p>
+
+The existing library for the Gamate doesn't implement C file
+I/O. There are no hacks for the <tt/read()/ and <tt/write()/ routines.
+
+To be more concrete, this limitation means that you cannot use any of the
+following functions (and a few others):
+
+<itemize>
+<item>printf
+<item>fclose
+<item>fopen
+<item>fread
+<item>fprintf
+<item>fputc
+<item>fscanf
+<item>fwrite
+<item>...
+</itemize>
+
+<sect>Other hints<p>
+
+<itemize>
+<item>The Gamate is emulated by MESS (<url url="http://www.mess.org/">),
+run like this: <tt>mess gamate -debug -window -skip_gameinfo -cart test.bin</tt>
```

Comment:
Okay, I start to understand what you mean with "obscure".

One of the non-written rules for new target in cc65@GitHub is that there's a reasonable free emulator supporting the target allowing "everybody" to verify it's functional correctness.

With (a non-release) MESS supporting Gamate I see that fullfilled - and as far as I understand it's not unusual for MESS to not include necessary ROMs (usually of IP reasons).